src/framework/core/js/FluidIoC.js

```diff
@@ -1812,6 +1887,20 @@ var fluid_3_0_0 = fluid_3_0_0 || {};
         return togo;
     };
 
+    fluid.findWorkflowShadows = function (shadows, blockStart, blockEnd, workflowRecord) {
+        var workflowShadows = [];
+        for (var i = blockStart; i < blockEnd; ++i) {
```

Comment:
Can this use `fluid.forEachInRange`

Reply:
Yes, I guess it could - in practice I only implemented fluid.forEachInRange to get around a tight corner with scoping issues where I was binding to what should be variables scoped to the loop in the various promises, and didn't want to use "let" since it is not properly supported in IE11. But now we've paid the expense of writing the 4-line function, we may as well use it : P